src/Contract/Operation/Associateable.php

```diff
@@ -13,17 +13,19 @@
 use loophp\collection\Contract\Collection;
 
 /**
- * @psalm-template TKey
- * @psalm-template TKey of array-key
- * @psalm-template T
+ * @template TKey
+ * @template T
  */
 interface Associateable
 {
     /**
-     * @psalm-param null|callable(TKey, TKey, T, Iterator<TKey, T>):(T|TKey) $callbackForKeys
-     * @psalm-param null|callable(T, TKey, T, Iterator<TKey, T>):(T|TKey) $callbackForValues
+     * @template U
+     * @template V
      *
-     * @psalm-return \loophp\collection\Collection<TKey|T, T|TKey>
+     * @param null|callable(TKey, TKey, T, Iterator<TKey, T>): U $callbackForKeys
+     * @param null|callable(T, TKey, T, Iterator<TKey, T>): V $callbackForValues
+     *
+     * @return \loophp\collection\Collection<TKey|U, T|V>
```

Comment:
the previous typing was more restrictive than what the operation actually allows you to do, because you can change the type of the keys and the values through the callbacks. However, this freedom is proving very difficult to type correctly and satisfy the analysers 😄

Reply:
Definitely.